fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt

```diff
@@ -865,6 +866,39 @@ class DefaultTabsTrayControllerTest {
         verify { controller.handleNavigateToBrowser() }
     }
 
+    @Test
+    fun `GIVEN a private tab is open and selected with a normal tab also open WHEN the private tab is closed and private home page shown and normal tab is selected from tabs tray THEN normal tab is displayed  `() {
+        val normalTab = TabSessionState(
+            content = ContentState(url = "https://simulate.com", private = false),
+            id = "normalTab",
+        )
+        val privateTab = TabSessionState(
+            content = ContentState(url = "https://mozilla.com", private = true),
+            id = "privateTab"
+        )
+        browserStore = spyk(BrowserStore(
+            initialState = BrowserState(
+                tabs = listOf(normalTab, privateTab),
+                selectedTabId = privateTab.id,
+            ),
+        ))
+        browsingModeManager.mode = BrowsingMode.Private
+        val controller = spyk(createController())
+        try {
+            mockkStatic("mozilla.components.browser.state.selector.SelectorsKt")
+            controller.handleTabSelected(privateTab, "source")
+            controller.handleTabDeletion("privateTab")
+            controller.handleTabSelected(normalTab, "source")
+
+            verify { controller.handleNavigateToBrowser() }
```

Comment:
I'm kinda confused what business logic path this test is covering. Is it supposed to be a unit test form of the STR (steps to reproduce) from the ticket?

Won't the verify always be true because of the first `controller.handleTabSelected(...)`? 

Based on the change to `handleTabSelected`, I was more expecting at least some simple coverage to bolster our confidence around the proposed logic change (i.e. `browsingModeManager.mode` was being set properly).